cmake/kokkos_functions.cmake

```
@@ -60,6 +60,235 @@ FUNCTION(kokkos_option CAMEL_SUFFIX DEFAULT TYPE DOCSTRING)
 
 ENDFUNCTION()
 
+FUNCTION(kokkos_append_config_line LINE)
+  SET(KOKKOS_TPL_EXPORTS "${KOKKOS_TPL_EXPORTS}\n${LINE}" PARENT_SCOPE)
+ENDFUNCTION()
+
+MACRO(kokkos_export_cmake_tpl NAME)
+  KOKKOS_APPEND_CONFIG_LINE("IF(NOT DEFINED ${NAME}_DIR AND NOT DEFINED ${NAME}_ROOT)")
+  KOKKOS_APPEND_CONFIG_LINE("  SET(${NAME}_DIR  ${${NAME}_DIR})")
+  KOKKOS_APPEND_CONFIG_LINE("  SET(${NAME}_ROOT ${${NAME}_ROOT})")
+  KOKKOS_APPEND_CONFIG_LINE("ENDIF()")
+  KOKKOS_APPEND_CONFIG_LINE("FIND_DEPENDENCY(${NAME})")
+ENDMACRO()
+
+MACRO(kokkos_export_imported_tpl NAME)
+  #make sure this also gets "exported" in the config file
+  KOKKOS_APPEND_CONFIG_LINE("ADD_LIBRARY(${NAME} UNKNOWN IMPORTED)")
+  KOKKOS_APPEND_CONFIG_LINE("SET_TARGET_PROPERTIES(${NAME} PROPERTIES")
+  
+  GET_TARGET_PROPERTY(TPL_COMPILE_OPTIONS ${NAME} INTERFACE_COMPILE_OPTIONS)
+  SET(TPL_LINK_OPTIONS)
+  IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.13.0")
+    GET_TARGET_PROPERTY(TPL_LINK_OPTIONS ${NAME} INTERFACE_LINK_OPTIONS)
+  ENDIF()
+  GET_TARGET_PROPERTY(TPL_LINK_LIBRARIES  ${NAME} INTERFACE_LINK_LIBRARIES)
+
+  GET_TARGET_PROPERTY(LIB_TYPE ${NAME} TYPE)
+  IF (NOT ${LIB_TYPE} STREQUAL "INTERFACE_LIBRARY")
+    GET_TARGET_PROPERTY(TPL_LIBRARY ${NAME} IMPORTED_LOCATION)
+  ENDIF()
+  GET_TARGET_PROPERTY(TPL_INCLUDES ${NAME} INTERFACE_INCLUDE_DIRECTORIES)
+
+  IF(TPL_LIBRARIES)
```

Should that be `TPL_LIBRARY`?
```suggestion
  IF(TPL_LIBRARY)
```

---

Yes, fixed.